File: base-programming/leetcode-00001-twoSum/00001-twoSum.go
package twosum

func twoSum(nums []int, target int) []int {
	deductedAddition := map[int]int{}

	for i, num := range nums {
		additionalToCurrentNumber := target - num

		j, ok := deductedAddition[additionalToCurrentNumber]
		if ok {
			return []int{j, i}
		}

		deductedAddition[num] = i
	}

	return nil
}

File: base-programming/leetcode-00001-twoSum/00001-twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const deductedAddition = new Map();

    for (let i = 0; i < nums.length; i++) {
        const additionToCurrentNumber = target - nums[i];

        if (deductedAddition.has(additionToCurrentNumber)) {
            return [i, deductedAddition.get(deducted)];
        }

        deductedAddition.set(nums[i], i);
    }

    return [-1, -1];
};
